pkg/resource/cluster/hook.go

```diff
@@ -313,28 +328,64 @@ func (rm *resourceManager) customUpdate(
 		}
 
 		if err := rm.associateEncryptionConfig(ctx, desired); err != nil {
-			awserr, ok := ackerr.AWSError(err)
-			// Check to see if we've raced an async update call and need to
-			// requeue
-			if ok && awserr.Code() == "ResourceInUseException" {
+			awsErr, ok := extractAWSError(err)
+
+			// Check to see if we've raced an async update call and need to requeue
+			if ok && awsErr.Code == "ResourceInUseException" {
 				return nil, requeueAfterAsyncUpdate()
 			}
 
 			return nil, err
 		}
 		// This doesn't reflect the actual status of the cluster, so we have to explicitly
 		// requeue and set the status to updating.
-		updatedRes.ko.Status.Status = aws.String(string(svcsdk.ClusterStatusUpdating))
+		updatedRes.ko.Status.Status = aws.String(string(svcsdktypes.ClusterStatusUpdating))
 		return returnClusterUpdating(updatedRes)
 	}
 
+	// Handle version updates
 	if delta.DifferentAt("Spec.Version") {
 		if err := rm.updateVersion(ctx, desired, latest); err != nil {
-			awserr, ok := ackerr.AWSError(err)
+			awsErr, ok := extractAWSError(err)
+
+			// Check to see if we've raced an async update call and need to requeue
+			if ok && awsErr.Code == "ResourceInUseException" {
+				return nil, requeueAfterAsyncUpdate()
+			}
+
+			return nil, err
+		}
+		return returnClusterUpdating(updatedRes)
+	}
+
+	// Handle computeConfig updates
+	if delta.DifferentAt("Spec.ComputeConfig") || delta.DifferentAt("Spec.StorageConfig") || delta.DifferentAt("Spec.KubernetesNetworkConfig") {
+		if err := rm.updateComputeConfig(ctx, desired); err != nil {
+			awsErr, ok := extractAWSError(err)
+			rlog.Info("attempting to update AutoMode config",
+				"error", err,
+				"isAWSError", ok,
+				"awsErrorCode", awsErr.Code)
 
```

Comment:
nit: we don't need to log here, there is a trace inside `updateComputeConfig` that'll do that for us.